src/Entities/ExerciseResult.cc

```diff
@@ -0,0 +1,46 @@
+#include "ExerciseResult.hh"
+
+#include <optional>
+
+namespace Entities
+{
+    Test::Test(const std::string &name, bool success)
+        : name_(name)
+    {
+        if (success)
+        {
+            errorMessage_ = std::nullopt;
+        }
+        else
+        {
+            errorMessage_ = "";
+        }
```

Comment:
I know this will probably change in the future but still.

```suggestion
        if (!success)
        {
            errorMessage_ = "";
        }
```
`std::nullopt` will be the default when initializing the class with `{}`.

Reply:
I feel making it explicit makes the code more readable.
When reading those lines, it is obvious that on success, the errrorMessage_ is empty, which is the reason we dont have a "success" attribute.